homeassistant/components/roborock/config_flow.py

```diff
@@ -0,0 +1,100 @@
+"""Config flow for Roborock."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from roborock.api import RoborockClient
+from roborock.containers import UserData
+from roborock.exceptions import RoborockException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_BASE_URL, CONF_ENTRY_CODE, CONF_USER_DATA, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RoborockFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Roborock."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._username: str | None = None
+        self._client: RoborockClient | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            username = user_input[CONF_USERNAME]
+            await self.async_set_unique_id(username.lower())
+            self._abort_if_unique_id_configured()
+            self._username = username
+            try:
+                _LOGGER.debug("Requesting code for Roborock account")
+                client = RoborockClient(username)
+                self._client = client
+                await client.request_code()
```

Comment:
Worth saving 1 line:

```suggestion
                self._client = RoborockClient(username)
                await self._client.request_code()
```